src/Symfony/Component/Messenger/Bridge/Amqp/Transport/Connection.php

```diff
@@ -566,4 +573,23 @@ private function getRoutingKeyForMessage(?AmqpStamp $amqpStamp): ?string
     {
         return $amqpStamp?->getRoutingKey() ?? $this->getDefaultPublishRoutingKey();
     }
+
+    private function withConnectionExceptionRetry(callable $callable): void
+    {
+        $maxRetries = 3;
+        $retries = 0;
+
+        retry:
+        try {
+            $callable();
+        } catch (\AMQPConnectionException $e) {
+            if (++$retries <= $maxRetries) {
+                $this->clear();
+
+                goto retry;
```

Comment:
This will cost years of public bashing, but I guess we can afford it 🥲

Reply:
:cupid: goto, no shame :)

Reply:
hehehe I like it too. I feel like imo it reads better than a loop specifically for retry logic.